### PR TITLE
Connect Contentful and Refactor State flow

### PR DESCRIPTION
**Pull request by McCambley**
Hey All,
I've got some videos that I'll let do most of the talking here, but here's a quick summary of what's included here. 
- Organized project data on Contentful with the help of Rebecca
- Used ApolloClient and GraphQl to fetch data from Contentful
- Lifted current course state and course update functionality up to `App` level so as to create a more consistent flow of information
- Lifted form toggle functionality and open state from `Homepage` to `App` to allow `ProfessionsPage` to access those as well.  

Recordings:
1) [How to connect to Contentful & a quick discussion of what tech was used and why](https://www.loom.com/share/1f1a3f7639da43ac825cadb4987ac0c5).
    - The beginning of video is a must watch, as it details how to use Tokens to connect with the CMS 
    - **What I forgot to mention was that your `.env` file must be included in your `.gitignore`**. It probably is by default, but just make sure. The world will keep spinning if you push those tokens to github, as this is a fairly casual project and our CMS is read only, it's just best to avoid that when possible.  
    - The latter portion of this video is for education. No need to watch it all unless you're curious.
    - [Here's a resource](https://www.howtographql.com/basics/1-graphql-is-the-better-rest/) on why to use GraphQl instead of a regular RESTapi
    - [Here's a resource](https://www.youtube.com/watch?v=mKPe6WB-dAc) on how to connect all the technologies used in our project
 2) [Live Demo](https://www.loom.com/share/57946158428f4b27a25131a22cc0f684)
    - This one is fun and shows how the code actually appears in the browser. Feel free to watch, or play around on your own. 
 3) [Explanation of refactoring](https://www.loom.com/share/23c1803fd0a1457ca7237e8f1ae683ab?sharedAppSource=personal_library)
    - I'd recommend watching this, as I talk about how state and functionality flow around this current iteration of the project. 
    - [Here's a link](https://miro.com/welcomeonboard/Ym9OYXNhNUpOQ1REb0ZqQjRHY2tjWVRhZUM4TDJ2VWhvMXRoaWNUUDkyUnc3MGlYMFp3dE1rNWF4V2hnd2RLQ3wzMDc0NDU3MzU1NDU5NzYwMDcx) to the Miro board I've been referencing in order to keep my brain in one piece
 4) [Code Walkthrough](https://www.loom.com/share/55495019b86246fab08a557aeaf543f1?sharedAppSource=personal_library)
    - I'll got through the actual code in this video. There's a few new things but it's mainly just a different way to conceptualize the above. 
    - This isn't necessary viewing, but might be helpful before diving into the code yourself. 

That's it! 🎉 Thanks for taking the time to read all that and watching what you could, I know it was a lot. 

Please don't let the length of this PR make it seem like I've got this all on lock. I'm hoping you all might catch a few things I missed here and there. Below is a list of things I'm thinking about doing when I've got some time, feel free to add to it! 

Y'all rock ✌️

**Future Concerns**
- Slightly cleaner loading UX on projects list
- It _is_ possible to get all courses at once. At one point it was deemed to be too _expensive_ by GraphQl, but I learned some things and that might make room for one initial larger fetch and no re-fetches._

### CODE REVIEW

File: src/components/projects/Projects.css [hunk 1]
```diff
@@ -2,6 +2,33 @@
   border: 1px solid red;
 } */
 
+/* LOADING STYLES */
+
+@keyframes shimmer {
```
Comment (denisehung) [inline src/components/projects/Projects.css:7]: Cool animation!

File: src/components/App.js [hunk 2]
```diff
@@ -3,20 +3,45 @@
...
+
   function handleNavClick() {
     setIsMenuOpen(!isMenuOpen);
   }
 
+  function handleCourseClick(queryString) {
+    setCourseQuery(queryString);
+  }
+
+  function handleButtonClick() {
```
Comment (denisehung) [inline src/components/App.js:37]: Good you moved this to App.js and connected it to both Homepage and Profession Page!

File: src/components/projects/Projects.css (hunk 1 above)
Comment (thinklikeadesigner) [reply src/components/projects/Projects.css:7]: I liked it too
I wonder if the one you were talking about, jake, just has some skinny horizontal containers to make it look like text

Comment (McCambley) [reply src/components/projects/Projects.css:7]: Me too! I like those text bars, I'll work on that!

**Pull request merged**